### PR TITLE
Add handler for verifying access_token

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-2035)

### CODE REVIEW

File: directory_api_client/company.py [hunk 1]
```diff
@@ -89,6 +90,11 @@ def verify_with_code(self, sso_session_id, code):
             self.endpoints['verify'], data, sso_session_id=sso_session_id
         )
 
+    def verify_with_companies_house(self, sso_session_id, access_token):
+        data = {'access_token': access_token}
```
Comment (konradko) [inline directory_api_client/company.py]: is this gonna be `access_token` or authorization code? Would be good to stick to oauth2 naming conventions

File: directory_api_client/company.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ class CompanyAPIClient(BaseAPIClient):
         'case-study-list': '/supplier/company/case-study/',
         'validate-company-number': '/validate/company-number/',
         'verify': '/supplier/company/verify/',
+        'verify-companies-house': '/supplier/company/verify/companies-house/',
```
Comment (konradko) [inline directory_api_client/company.py]: not a huge fan of this url structure - makes ch-verify looks like a sub-category of the general verify. Maybe `/supplier/company/verify-companies-house/`?

File: directory_api_client/company.py (hunk 1 above)
Comment (richtier) [inline directory_api_client/company.py]: the variable name was inspired by the CH oauth2 docs:

image

Comment (konradko) [inline directory_api_client/company.py]: yes but this is not what you will get from FAB - in the callback you get authorization code, that you exchange for an access token, that should happen in API:

```
```
POST /oauth2/token HTTP/1.1

grant_type=authorization_code&
code=authorization-code&
client_id=your-client-id&
client_secret=your-client-secret&
redirect_uri=application-redirect-uri
```
```

Comment (konradko) [inline directory_api_client/company.py]: ok, I see FAB will be doing the authorization code > access token exchange and pass that to API

**Pull request merged**